benchmarks/checkpointing.ts

```diff
@@ -1,54 +1,20 @@
 import { pseudoRandomBytes } from 'crypto'
 import { CheckpointTrie } from '../dist'
 
-const iterations = 5000
-const samples = 5
+export const iterTest = async (numOfIter: number) => {
+  const keys: Buffer[] = []
+  const vals: Buffer[] = []
 
-const iterTest = async (numOfIter: number): Promise<Array<number>> => {
-  return new Promise(async (resolve) => {
-    let vals = [] as any
-    let keys = [] as any
-
-    for (let i = 0; i <= numOfIter; i++) {
-      vals.push(pseudoRandomBytes(32))
-      keys.push(pseudoRandomBytes(32))
-    }
-
-    let hrstart = process.hrtime()
-    let numOfOps = 0
-    let trie = new CheckpointTrie()
-    for (let i = 0; i < numOfIter; i++) {
-      trie.checkpoint()
-      await trie.put(vals[i], keys[i])
-      await trie.get(Buffer.from('test'))
-      numOfOps++
-      if (numOfOps === numOfIter) {
-        const hrend = process.hrtime(hrstart)
-        resolve(hrend)
-      }
-      trie.commit()
-    }
-  })
-}
+  for (let i = 0; i <= numOfIter; i++) {
+    keys.push(pseudoRandomBytes(32))
+    vals.push(pseudoRandomBytes(32))
+  }
 
-const go = async () => {
-  let i = 1
-  let avg = [0, 0]
+  const trie = new CheckpointTrie()
 
-  console.log(`Benchmark 'checkpointing' starting...`)
-  while (i <= samples) {
-    console.log(`Sample ${i} with ${iterations} iterations.`)
-    const hrend = await iterTest(iterations)
-    avg[0] += hrend[0]
-    avg[1] += hrend[1]
-    // console.log('benchmarks/checkpointing.ts | execution time: %ds %dms', hrend[0], (hrend[1] / 1000000).toFixed(3))
-    i++
+  for (let i = 0; i < numOfIter; i++) {
+    trie.checkpoint()
+    await trie.put(keys[i], vals[i])
```

Comment:
Something else that'd be interesting to see is how checkpointing performance changes with the size of data to be commited. Here that size is one value (each value is committed separately).

Reply:
Something else that'd be interesting for finding the slowdown is a benchmark that does `checkpoint`, inserts a bunch of stuff and then does `revert` instead of `commit`.

Comparing this bench on v4 and v3 should tell us whether the slowdown is happening in `commit` (which is my suspicion) or not.